### PR TITLE
Create circular beam profile

### LINKED ISSUE #29001: Create circular beam profile
The beam profile affects the results of the Mantid Monte Carlo absorption correction calculations. Mantid currently only supports a rectangular beam profile. For example if you look at the SetBeam algorithm the only option accepted for the shape is "Slit":

Chris Ridley has requested that we add support for a circular beam profile because they have a circular slit as part of the Pearl instrument environment which makes the beam incident on the sample circular.

The implementation of this should follow the RectangularBeamProfile class ie create child of IBeamProfile called CircularBeamProfile and implement a generatePoint function that randomly selects a point within the circular cross section of the beam. There are a couple of ways of achieving the even distribution of points across the circle (throwaway or can evenly generate them based on polar coordinates)

Once the CircularBeamProfile class has been created it needs supporting in the SetBeam algorithm by providing a new Shape option and a radius parameter. It would be nice to also support it in the Instrument Definition Files but that can be done separately

### PR DESCRIPTION
**Pull request by clayton-tom**
**Description of work.**
Create a Circular Beam profile and provide support in the SetBeam, MonteCarlo and AddAbsorptionWeightedPathLengths algorithms.

**Report to:** Danny Hindson, Chris Ridley

**To test:**

[29001_test.zip](https://github.com/mantidproject/mantid/files/5109407/29001_test.zip)

The attached script runs a MonteCarloAbsorption using both a Slit (0.5x0.5) and Circular (radius=0.5) beam profile, with two separate output workspaces.

1. Download and run the test script
2. Plot and overplot the two output workspaces
3. Check that there are small differences in the results but that the plot shape remains mostly similar

Fixes #29001.

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Review by Harrietbrown — DISMISSED** [review]
The code runs as described, spectra in output workspaces have a similar shape. The code looks good and the tests are well written. Good job.

File: Testing/SystemTests/tests/analysis/PaalmanPingsMonteCarloAbsorptionTest.py [hunk 1]
```diff
@@ -44,6 +44,7 @@ def runTest(self):
             ContainerDensity=6.0,
             CorrectionsWorkspace='flat_plate_corr'
         )
+        SaveNexus(InputWorkspace='flat_plate_corr', Filename='C:\\NewTestData\\CLAYTONirs_PP_MC_flat_plate.nxs')
```
Comment (martyngigg) [inline Testing/SystemTests/tests/analysis/PaalmanPingsMonteCarloAbsorptionTest.py:47]: Can I ask why we are now saving to NeXus as part of these tests? If this is necessary can we avoid hardcoding OS-dependent paths?

File: Framework/Algorithms/inc/MantidAlgorithms/SampleCorrections/CircularBeamProfile.h [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+  IBeamProfile::Ray
+  generatePoint(Kernel::PseudoRandomNumberGenerator &rng) const override;
+  IBeamProfile::Ray
+  generatePoint(Kernel::PseudoRandomNumberGenerator &rng,
+                const Geometry::BoundingBox &bounds) const override;
+  Geometry::BoundingBox
+  defineActiveRegion(const Geometry::BoundingBox &sampleBox) const override;
+
+private:
+  // const double m_azimuthDeg;
```
Comment (martyngigg) [inline Framework/Algorithms/inc/MantidAlgorithms/SampleCorrections/CircularBeamProfile.h:39]: Can we remove the commented out code?

**Review by martyngigg — CHANGES_REQUESTED** [review]
Just a question surrounding the updates to the system tests.

File: Testing/SystemTests/tests/analysis/PaalmanPingsMonteCarloAbsorptionTest.py (hunk 1 above)
Comment (clayton-tom) [reply Testing/SystemTests/tests/analysis/PaalmanPingsMonteCarloAbsorptionTest.py:47]: I added these lines to save a copy of what would be the new reference files locally, but hadn't realised I'd committed them - I'll get rid

**Comment by martyngigg** [comment]
Unrelated build failure.

**Review by martyngigg — APPROVED** [review]

**Pull request merged**